### PR TITLE
MET-2857 Create endpoint to execute depublishing bypassing the workfl…

### PR DESCRIPTION
**Pull request by stzanakis**
…ow creation

### CODE REVIEW

File: metis-common/src/main/java/eu/europeana/metis/RestEndpoints.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
   // DEPUBLISHED RECORDS
   public static final String DEPUBLISH_RECORDIDS_DATASETID = "/depublish/record_ids/{datasetId}";
+  public static final String DEPUBLISH_RECORDIDS_EXECUTE_DATASETID = "/depublish/record_ids/execute/{datasetId}";
```
Comment (jochen-vermeulen) [inline metis-common/src/main/java/eu/europeana/metis/RestEndpoints.java:32]: Just a small issue perhaps, but I don't think that this path is appropriate, given that the task being executed may not be a depublication for individual records. What about `/depublish/execute/{datasetId}`?

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/OrchestratorService.java [hunk 2]
```diff
@@ -288,14 +290,14 @@ public WorkflowExecution getWorkflowExecutionByExecutionId(MetisUser metisUser,
    * </ul>
    */
   public WorkflowExecution addWorkflowInQueueOfWorkflowExecutionsWithoutAuthorization(
-      String datasetId, ExecutablePluginType enforcedPredecessorType, int priority)
+      String datasetId, Workflow workflow, ExecutablePluginType enforcedPredecessorType, int priority)
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/OrchestratorService.java:293]: Rename to match the name in the method below (`workflowProvided`)? Or perhaps something else (e.g. `alternativeBaseWorkflow`)?

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java [hunk 3]
```diff
@@ -178,4 +189,24 @@ public int addRecordIdsToBeDepublished(MetisUser metisUser, String datasetId,
     result.setResultsAndLastPage(records, depublishRecordIdDao.getPageSize(), page);
     return result;
   }
+
+  public WorkflowExecution createDepublishWorkflowExecution(MetisUser metisUser, String datasetId,
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java:193]: Somewhat misleading method name? Perhaps something like `scheduleDepublishWorkflowExecution` to make clear that it is not just creating but also executing?

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java [hunk 4]
```diff
@@ -178,4 +189,24 @@ public int addRecordIdsToBeDepublished(MetisUser metisUser, String datasetId,
...
+    //Prepare depublish workflow, do not save in the database. Only create workflow execution
+    final Workflow workflow = new Workflow();
+    workflow.setDatasetId(datasetId);
+    final DepublishPluginMetadata depublishPluginMetadata = new DepublishPluginMetadata();
+    depublishPluginMetadata.setEnabled(true);
+    depublishPluginMetadata.setDatasetDepublish(datasetDepublish);
+    workflow.setMetisPluginsMetadata(Collections.singletonList(depublishPluginMetadata));
+
+    // Validate the new workflow.
+    workflowUtils.validateWorkflowPlugins(workflow, null);
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java:207]: This is not necessary I think. This already happens when calling `orchestratorService.addWorkflowInQueueOfWorkflowExecutions`?

File: metis-common/src/main/java/eu/europeana/metis/RestEndpoints.java (hunk 1 above)
Comment (stzanakis) [reply metis-common/src/main/java/eu/europeana/metis/RestEndpoints.java:32]: Good point.

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/OrchestratorService.java (hunk 2 above)
Comment (stzanakis) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/OrchestratorService.java:293]: ok

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java (hunk 3 above)
Comment (stzanakis) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java:193]: Renamed

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java (hunk 4 above)
Comment (stzanakis) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java:207]: Indeed! Removed it

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java [hunk 5]
```diff
@@ -203,9 +203,6 @@ public WorkflowExecution createDepublishWorkflowExecution(MetisUser metisUser, S
     depublishPluginMetadata.setDatasetDepublish(datasetDepublish);
     workflow.setMetisPluginsMetadata(Collections.singletonList(depublishPluginMetadata));
 
-    // Validate the new workflow.
-    workflowUtils.validateWorkflowPlugins(workflow, null);
-
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java]: This change now made the workflowUtils variable in this class unused. Best to remove it, right?

Comment (stzanakis) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java]: Indeed, I missed that

**Pull request merged**